examples_tests/06.MeshLoaders/main.cpp

```diff
@@ -150,59 +326,41 @@ int main()
                         core::matrix3x4SIMD MV = camera->getViewMatrix();
                         memcpy(uboData.data() + shdrIn.descriptorSection.uniformBufferObject.relByteoffset, MV.pointer(), shdrIn.descriptorSection.uniformBufferObject.bytesize);
-                    }
-                    break;
+                    } break;
+                    
                     case asset::IRenderpassIndependentPipelineMetadata::ECSI_WORLD_VIEW_INVERSE_TRANSPOSE:
                     {
                         core::matrix3x4SIMD MV = camera->getViewMatrix();
                         memcpy(uboData.data()+shdrIn.descriptorSection.uniformBufferObject.relByteoffset, MV.pointer(), shdrIn.descriptorSection.uniformBufferObject.bytesize);
-                    }
-                    break;
+                    } break;
                 }
             }
         }       
-        driver->updateBufferRangeViaStagingBuffer(gpuubo.get(), 0ull, gpuubo->getSize(), uboData.data());
+
+        commandBuffer[0]->updateBuffer(gpuubo.get(), 0ull, gpuubo->getSize(), uboData.data());
 
         for (auto gpumb : gpumesh->getMeshBuffers())
         {
-            const video::IGPURenderpassIndependentPipeline* pipeline = gpumb->getPipeline();
+            const video::IGPURenderpassIndependentPipeline* gpuRenderpassIndependentPipeline = gpumb->getPipeline();
             const video::IGPUDescriptorSet* ds3 = gpumb->getAttachedDescriptorSet();
+            
+            commandBuffer[0]->bindGraphicsPipeline(gpuGraphicsPipeline.get()); // I'm wondering if I should create gpuGraphicsPipeline for each mesh buffer before
 
-            driver->bindGraphicsPipeline(pipeline);
             const video::IGPUDescriptorSet* gpuds1_ptr = gpuds1.get();
-            driver->bindDescriptorSets(video::EPBP_GRAPHICS, pipeline->getLayout(), 1u, 1u, &gpuds1_ptr, nullptr);
+            commandBuffer[0]->bindDescriptorSets(asset::EPBP_GRAPHICS, gpuRenderpassIndependentPipeline->getLayout(), 1u, 1u, &gpuds1_ptr, nullptr);
             const video::IGPUDescriptorSet* gpuds3_ptr = gpumb->getAttachedDescriptorSet();
             if (gpuds3_ptr)
-                driver->bindDescriptorSets(video::EPBP_GRAPHICS, pipeline->getLayout(), 3u, 1u, &gpuds3_ptr, nullptr);
-            driver->pushConstants(pipeline->getLayout(), video::IGPUSpecializedShader::ESS_FRAGMENT, 0u, gpumb->MAX_PUSH_CONSTANT_BYTESIZE, gpumb->getPushConstantsDataPtr());
+                commandBuffer[0]->bindDescriptorSets(asset::EPBP_GRAPHICS, gpuRenderpassIndependentPipeline->getLayout(), 3u, 1u, &gpuds3_ptr, nullptr);
+            commandBuffer[0]->pushConstants(gpuRenderpassIndependentPipeline->getLayout(), video::IGPUSpecializedShader::ESS_FRAGMENT, 0u, gpumb->MAX_PUSH_CONSTANT_BYTESIZE, gpumb->getPushConstantsDataPtr());
 
-            driver->drawMeshBuffer(gpumb);
+            // commandBuffer[0]->draw(vertexCount, instanceCount, firstVertex, firstInstance); // TODO! Damn, do I need an access to VAO and vertex stuff?
```

Comment:
what about this one?

Reply:
implement as `inline` method (utility) in `ICommandBuffer`, look at the original `I[Video?]Driver::drawMeshBuffer`

Reply:
sure